Use Spree::ActiveShipping::Config

swcraig opened the pull request:
After solidusio/solidus@ec87854 static preferences have become the default. This means that we should use the global instance of `Spree::ActiveShipping::Config` to set and retrieve our preferences. The specs were failing because we were using the old way of persisting preferences via the database (which has become an opt-in legacy option). Instead of taking the legacy approach to preferences, this PR updates the code to use the global instance of `Spree::ActiveShipping::Config` that is assigned in the initializer, instead of creating new instances of the preferences each time.

**EDIT**: This PR also adds ffaker as a development dependency in the gemspec. It was removed as a runtime dependency of Solidus.

swcraig committed:
Use Spree::ActiveShipping::Config in controller

After solidusio/solidus@ec87854 in Solidus core, using a global static
preferences model (rather than a database backed one) is the default.
This means that we should use the global instance of
Spree::ActiveShipping::Config to store and retrieve the preferences. The
new instance of the preferences that was created in the controller was
not persisting the changes since there was no longer a database backing
it.

From now on we should use the global instance of
Spree::ActiveShipping::Config that is assigned in the initializer.

swcraig committed:
Use Spree::ActiveShipping::Config in specs

After solidusio/solidus@ec87854 in Solidus core, using a global static
preferences model (rather than a database backed one) is the default.
This means that we should use the global instance of
Spree::ActiveShipping::Config to store and retrieve the preferences. The
new instances of the preferences that were created in the specs were
not persisting the changes since there was no longer a database backing
them.

From now on we should use the global instance of
Spree::ActiveShipping::Config that is assigned in the initializer.

forkata commented:
@swcraig Thank you for fixing this! Can you add a PR description and some comments to the commits to add some reasoning for why we're switching to `Spree::ActiveShipping::Config`. Other than that, all looks good to me!

swcraig committed:
Add ffaker development dependency to gemspec

ffaker was removed as a runtime dependency of Solidus here:
solidusio/solidus#2163

forkata commented:
Thanks @swcraig, great work :tada:
